src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -710,7 +710,7 @@ private static void ExcludeFiles(List<PhysicalPackageFile> searchFiles, string b
 
         private static void CreatePart(ZipArchive package, string path, Stream sourceStream)
         {
-            if (PackageHelper.IsNuspec(path) || ProjectJsonPathUtilities.IsProjectConfig(path))
```

Comment:
Yes, it allows the project.json to be packed in the package if you've specifically included it.  We block including the nuspec file because we find all files and don't want it to get picked up with all of the other files.  For project.json, we don't do that * include so it won't get included unless you want it.